### PR TITLE
Update workflow as it has developed in recent projects

### PR DESCRIPTION
**Pull request by EnnioWolsink**
This is hopefully everything! Not sure how much you want to keep of what's in /workflow/config/params.sh.dist.

Do with this as you please :)

### CODE REVIEW

File: workflow/config/config.sh [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
+    local path_project="${path_current}/../../.."
+    
+    . "${path_wf}/util/bash_setup.sh"
+    . "${path_wf}/util/printing.sh"
+    . "${path_wf}/util/configuration.sh"
+    
+    
+    # Defaults
+    
+    config_editor=nano
```
Comment (mkrause) [inline workflow/config/config.sh]: vim please ;)

File: workflow/config/params.sh.dist [hunk 2]
```diff
@@ -1,4 +1,23 @@
-# Local parameters
+# Machine-local configuration
```
Comment (mkrause) [inline workflow/config/params.sh.dist]: This needs to be moved to `.env.example`.

File: workflow/workflow.sh [hunk 3]
```diff
@@ -53,28 +53,39 @@ EOT
...
 
 # --------------------------------
 # Basic commands
 # --------------------------------
 
+command_names+=("self-installed")
+command_descriptions+=("Whether workflow has been installed")
+cmd_self_installed() {
+    local path_env="${path_root}/.env"
+    local path_env_dist="${path_root}/.env.example.dist"
```
Comment (mkrause) [inline workflow/workflow.sh]: `.env.example.dist` needs to be `.env.example`. The latter is the standard name, certain tooling won't pick it up if it's named differently.

File: workflow/workflow.sh [hunk 4]
```diff
@@ -53,28 +53,39 @@ EOT
...
+}
+
 command_names+=("self-install")
 command_descriptions+=("Install workflow")
 cmd_self_install() {
     local force=0
-    local path_params="${path_script}/config/params.sh"
-    local path_params_dist="${path_script}/config/params.sh.dist"
+    local path_env="${path_root}/.env"
+    local path_env_dist="${path_root}/.env.example.dist"
```
Comment (mkrause) [inline workflow/workflow.sh]: Same.

**Review by mkrause — CHANGES_REQUESTED** [review]

**Comment by EnnioWolsink** [comment]
Processed requested changes :)

**Review by mkrause — APPROVED** [review]

**Comment by mkrause** [comment]
Nice work! Thanks for the update.

**Pull request merged**